src/interp/binary-reader-interp.cc

```diff
@@ -1169,6 +1172,9 @@ Result BinaryReaderInterp::OnReturnCallExpr(Index func_index) {
 
 Result BinaryReaderInterp::OnReturnCallIndirectExpr(Index sig_index,
                                                     Index table_index) {
+  if (sig_index >= module_.func_types.size()) {
+    return Result::Error;
+  }
```

Comment:
I wonder if these checks should be in the validator instead?

Reply:
Actually it looks like we already have this code above, can we call this instead?

 ```
CHECK_RESULT(
      validator_.OnCall(GetLocation(), Var(func_index, GetLocation())));		    
```

Reply:
Ah, good point. I changed them to use validator_.OnReturnCall and validator_.OnReturnCallIndirect which looks like what it should have been doing all along.